src/main/java/org/opensearch/knn/index/query/KNNQuery.java

```diff
@@ -69,6 +73,42 @@ public KNNQuery(
         this.parentsFilter = parentsFilter;
     }
 
+    /**
+     * Constructor for KNNQuery with query vector, index name and parent filter
+     *
+     * @param field field name
+     * @param queryVector query vector
+     * @param indexName index name
+     */
+    public KNNQuery(String field, float[] queryVector, String indexName, BitSetProducer parentsFilter) {
+        this.field = field;
+        this.queryVector = queryVector;
+        this.indexName = indexName;
+        this.parentsFilter = parentsFilter;
+    }
+
+    /**
+     * Constructor for KNNQuery with radius
+     *
+     * @param radius engine radius
+     * @return KNNQuery
+     */
+    public KNNQuery radius(Float radius) {
+        this.radius = radius;
+        return this;
+    }
```

Comment:
Because the current object construction chaining usage enhances readability and usability compared to plain setter. I don't see benefit to convert the whole class to builder pattern in this class. 

Current:
```
KNNQuery query = new KNNQuery(fieldName, vector, indexName, parentFilter).radius(radius);
```

Change with plain setter:
```
KNNQuery query = new KNNQuery(fieldName, vector, indexName, parentFilter);
query.radius(radius);
```